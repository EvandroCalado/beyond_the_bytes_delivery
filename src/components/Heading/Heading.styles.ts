import styled, { css } from 'styled-components';

import { HeadingProps } from '.';

export const Title = styled.h1<HeadingProps>`
  ${({ theme, size = 'md', fontWeight = 'normal' }) => css`
    font-size: ${theme.font.sizes[size]};
    font-weight: ${fontWeight};
  `}
`;
