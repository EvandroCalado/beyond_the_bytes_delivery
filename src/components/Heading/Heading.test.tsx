import { render, screen } from 'utils';

import theme from 'styles/theme';

import { Heading } from './Heading';

describe('Heading', () => {
  it('should render default', () => {
    render(<Heading>test</Heading>);

    const heading = screen.getByRole('heading', { level: 1 });

    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent('test');
    expect(heading).toHaveStyle({
      fontSize: theme.font.sizes.md,
    });
  });

  it('should render with as, size and fontWeight', () => {
    render(
      <Heading as="h2" size="lg" fontWeight="bold">
        test
      </Heading>,
    );

    const heading = screen.getByRole('heading', { level: 2 });

    expect(heading).toHaveStyle({
      fontSize: theme.font.sizes.lg,
      fontWeight: 'bold',
    });
  });

  it('should render a snapshot', () => {
    const { container } = render(<Heading>test</Heading>);

    expect(container).toMatchSnapshot();
  });
});
