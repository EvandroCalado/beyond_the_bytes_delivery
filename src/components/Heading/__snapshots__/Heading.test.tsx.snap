// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render a snapshot 1`] = `
.c0 {
  font-size: 1.6rem;
  font-weight: normal;
}

<div>
  <h1
    class="c0"
  >
    test
  </h1>
</div>
`;
